### DISCUSSION TITLE
Support Teamcenter application startup configuration

### DISCUSSION THREAD

**Discussion by x28358** [JaneWardSandy/eclipse-pde-partial-idea#79] [Q&A] [answered]
HI, I made an error in configuring TeamCenter startup. 
How can I solve this problem? 
Looking forward to your reply

![image](https://user-images.githubusercontent.com/59767564/170652811-3c6c996a-00eb-4794-abf9-6e2f2acba585.png)

**Comment by JaneWardSandy** [comment]
I haven't tested it on Teamenter 12 yet
But it's normal in Teamcenter 10 and 11
I'll cross-test Teamcenter 12 with Eclipse and IDEA soon to confirm the problem

**Comment by JaneWardSandy** [comment, marked as answer]
Which version of Teamcenter are you using?
Does Teamcenter work by itself?
Does debugging Teamcenter with Eclipse start normally?

I used Teamcenter 12.0 and 12.4 and it starts fine in both Eclipse and IDEA.

From the log prompt, this should be a version conflict of Teamcenter's own Bundle. It may be that the jar package under plugins was not cleaned up when your Teamcenter was upgraded.

When OSGI loads the bundle, a certain version of `org.eclipse.ui.intro.universal` has been loaded, and the bundle is marked as a `singleton`; but in the subsequent bundle loading process, another version of `org.eclipse.ui.intro.universal` is required to be loaded again, causing a singleton load violation.

So first try to start Teamcenter directly and use Eclipse to start Teamcenter to confirm that this is not a problem with the IDEA plugin.

**Reply by x28358** [reply]
Thank you very much for your reply. I also use TC12.4, and it works normally in Eclipse.

![3a9bb8e2081a0914b41348963b85cfd](https://user-images.githubusercontent.com/59767564/170930411-cc50006b-2ad6-44c3-87d2-aa0a7de73ffe.png)
![d8c601a192048bb5b68ee308ed82b2c](https://user-images.githubusercontent.com/59767564/170930432-0a9bf8bc-6325-4a00-aa71-fdd54a4d81ca.png)
![3929e53987d0dafd0003d2cd3f26777](https://user-images.githubusercontent.com/59767564/170930442-3ae0b021-a204-4544-818c-365c1138c8cc.png)

**Reply by x28358** [reply]
Can you write Chinese?  If you can, I am looking forward for your Chinese reply.

**Reply by JaneWardSandy** [reply]
Please remove the contents of the current `Target` and only add the `Teamcenter Portal` directory.  
请移除当前`Target`中的内容，仅添加`Teamcenter Portal`目录。  

The plugin supports detecting `teamcneter(.exe)` and `eclipse(.exe)` in the directory.  
插件支持检测目录下的`teamcneter(.exe)`和`eclipse(.exe)` 。 

Judging from your current `Target`, it should be that both Eclipse and Teamcenter contain `org.eclipse.ui.intro.universal`, and the versions are inconsistent and both are required to be loaded.  
从你当前的`Target`推测，应该是Eclipse和Teamcenter中均含有`org.eclipse.ui.intro.universal`，并且版本不一致且均被要求加载导致的冲突。

**Reply by x28358** [reply]
Problem solved. Thank you for your great help.

**Comment by x28358** [comment]
Hi, I have a new question. 
The project suddenly could not find FMS_HOME. 
It still works in Eclipse.
The configurations and errors are as follows.

![a1](https://user-images.githubusercontent.com/59767564/171583499-229114a9-8e83-4fd8-a385-a4ff1466864f.png)
![a2](https://user-images.githubusercontent.com/59767564/171583516-1ce077c0-1112-4975-8bda-ff0075de4fbb.png)

**Reply by JaneWardSandy** [reply]
Append `%FMS_HOME%/bin;%FMS_HOME%/lib;` to `path` on user / system environments,  or set it in run configuration's environment.
<img width="657" alt="image" src="https://user-images.githubusercontent.com/15345355/171596566-88ead4ad-a44c-4577-9762-7a86377f3c3b.png">

**Reply by x28358** [reply]
After following your configuration, this situation still exists, but after copying the new project, TeamCenter starts successfully, but the code is not loaded

**Reply by JaneWardSandy** [reply]
If you set `path=%path%;%FMS_HOME%/bin;%FMS_HOME%/lib` to user variables, you should exit `IDEA` completely, and then start `IDEA` to make the new environment variables take effect;

If you modify the system environment variables to append content, you should restart the system to reload the system environment variables;

If you are modifying the environment variables in the `run configuration`, consider checking the `Clean runtime directory` to clear the runtime folder;

As you said modules are not loaded, you should consider checking if `PDE Facet` is added for each module.
